shopify_integration/shopify_integration/doctype/shopify_settings/shopify_settings.js

```diff
@@ -48,33 +48,37 @@ frappe.ui.form.on("Shopify Settings", {
 			`);
 		}
 
-		frm.add_custom_button(__("Products"), function () {
-			frappe.call({
-				method: "shopify_integration.products.sync_products_from_shopify",
-				freeze: true,
-				callback: function (r) {
-					if (r.message) {
-						frappe.msgprint(__("Product sync has been queued. This may take a few minutes."));
-					} else {
-						frappe.msgprint(__("Something went wrong while trying to sync products. Please check the latest Shopify logs."))
+		if (frm.doc.enable_shopify) {
+			frm.add_custom_button(__("Products"), function () {
+				frm.call({
+					doc: frm.doc,
+					method: "sync_products",
+					freeze: true,
+					callback: function (r) {
+						if (r.message) {
+							frappe.msgprint(__("Product sync has been queued. This may take a few minutes."));
+						} else {
+							frappe.msgprint(__("Something went wrong while trying to sync products. Please check the latest Shopify logs."))
+						}
 					}
-				}
-			})
-		}, __("Sync"))
-
-		frm.add_custom_button(__("Payouts"), function () {
-			frappe.call({
-				method: "shopify_integration.payouts.sync_payouts_from_shopify",
-				freeze: true,
-				callback: function (r) {
-					if (r.message) {
-						frappe.msgprint(__("Payout sync has been queued. This may take a few minutes."));
-					} else {
-						frappe.msgprint(__("Something went wrong while trying to sync payouts. Please check the latest Shopify logs."))
+				})
+			}, __("Sync"));
+
+			frm.add_custom_button(__("Payouts"), function () {
+				frm.call({
+					doc: frm.doc,
+					method: "sync_payouts",
+					freeze: true,
+					callback: function (r) {
```

Comment:
Can we maybe have a date filter or something? I don't want to be fetching ALL payouts with a single button press.

Reply:
This actually won't pull all payouts (except maybe on the first pull), since it uses a "Last Sync" field as the start date.

For the first pull case, we can add a default number of days, or let them input a start date.

Reply:
@vjFaLk 

**Update**: I've completely forgotten this codebase lol. Apparently, I had already added a fallback to only pull payouts from the first day of the current month.